storage/engine/gc.go

```diff
@@ -88,7 +92,11 @@ func (gc *GarbageCollector) Filter(keys []proto.EncodedKey, values [][]byte) pro
 	// If there are no non-deleted survivors, return timestamp of first key
 	// to delete all entries.
 	if !survivors {
-		_, ts, _ := MVCCDecodeKey(keys[0])
+		_, ts, _, err := MVCCDecodeKey(keys[0])
+		if err != nil {
+			log.Errorf("unable to decode MVCC key: %q: %v", keys[0], err)
```

Comment:
`proto.ZeroTimestamp` indicates that nothing will be deleted, unless there's a caveat I'm missing.